enhancements/builds/node-cache-for-builds.md

```diff
@@ -0,0 +1,138 @@
+---
+title: node-cache-for-builds
+authors:
+  - "@adambkaplan"
+reviewers:
+  - "@nalind
+  - "@rhdan"
+  - "@TomSweeneyRedHat"
+  - "@bparees"
+approvers:
+  - "@bparees"
+creation-date: 2019-02-18
+last-updated: 2019-02-18
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Node Cache for Builds
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Can cri-o be enhanced so pods are not scheduled if the kubelet/node garbage collection is in progress?
```

Comment:
what's the motivation for the question?

and i think the answer is no, anyway.  crio does not control the scheduler.  at best you could have crio not *start* a container if GC was happening, maybe.

and you'd still have the problem of GC starting after the container started(which you definitely cannot block because then you'd potentially starve out GC from ever running), so i don't think it solves anything.

Reply:
No.  Kubelet directs CRI-O to do GC.

Reply:
I recall this coming up during our conversation with @mrunalp, @nalind, and @rhatdan back in December 2019. It would be nice if build pods didn't start while GC is in progress, but I agree that we still run the risk of caches being pruned during the build.

I can strike this as an open question.